Detections/SigninLogs/FailedLogonToAzurePortal.yaml

```diff
@@ -23,21 +24,38 @@ relevantTechniques:
 query: |
 
   let timeRange = 1d;
-  SigninLogs
+  let lookBack = 7d;
+  let threshold_Failed = 5;
+  let threshold_FailedwithSingleIP = 20;
+  let threshold_IPAddressCount = 2;
+  let azPortalSignins = SigninLogs
   | where TimeGenerated >= ago(timeRange)
-  | extend OS = DeviceDetail.operatingSystem, Browser = DeviceDetail.browser
-  | extend StatusCode = tostring(Status.errorCode), StatusDetails = tostring(Status.additionalDetails)
-  | where AppDisplayName contains "Azure Portal"
-  // Exclude non-failure types
-  | where ResultType !in ("0", "50125", "50140")
-  // Removing unresolved identities that don't provide status
-  | where isnotempty(StatusDetails) and isnotempty(Location) and not(Identity matches regex "[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}")
-  | extend Status = strcat(StatusCode, ": ", ResultDescription) 
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), Status = makelist(Status), StatusDetails = makelist(StatusDetails), IPAddresses = makelist(IPAddress), IPAddressCount = dcount(IPAddress)
-  by UserPrincipalName, AppDisplayName, tostring(Browser), tostring(OS), Location, CorrelationId  
-  | mvexpand IPAddresses, Status, StatusDetails
-  | extend Status = strcat(Status, "-", StatusDetails)
-  | summarize Status = makeset(Status) by StartTimeUtc, EndTimeUtc, tostring(IPAddresses), UserPrincipalName, AppDisplayName, tostring(Browser), tostring(OS), Location, IPAddressCount, CorrelationId
-  | where Status !has "MFA requirement satisfied by claim in the token"
-  | summarize Status = makeset(Status), IPAddresses = makeset(IPAddresses), CorrelationIds = makeset(CorrelationId) by StartTimeUtc, EndTimeUtc, UserPrincipalName, AppDisplayName, tostring(Browser), tostring(OS), Location, IPAddressCount
-  | extend timestamp = StartTimeUtc, AccountCustomEntity = UserPrincipalName, IPCustomEntity = tostring(IPAddresses)
+  // Azure Portal only and exclude non-failure Result Types
+  | where AppDisplayName has "Azure Portal" and ResultType !in ("0", "50125", "50140")
+  // Tagging identities not resolved to friendly names
+  | extend Unresolved = iff(Identity matches regex "[0-9a-z]{8}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{4}-[0-9a-z]{12}", true, false)
```

Comment:
May want to promote regex pattern to the top as variable as it is used at 2 places so can be changed at one central place if needed in the future

Reply:
done